flink-runtime/src/test/java/org/apache/flink/runtime/state/StateBackendTestBase.java

```diff
@@ -3747,6 +3747,39 @@ public void process(Integer key, ListState<String> state) throws Exception {
 		}
 	}
 
+	/**
+	 * Since {@link AbstractKeyedStateBackend#getKeys(String, Object)} does't support concurrent modification
+	 * and {@link AbstractKeyedStateBackend#applyToAllKeys(Object, TypeSerializer, StateDescriptor,
+	 * KeyedStateFunction)} rely on it to get keys from backend. So we need this unit test to verify the concurrent
+	 * modification with {@link AbstractKeyedStateBackend#applyToAllKeys(Object, TypeSerializer, StateDescriptor, KeyedStateFunction)}.
+	 */
+	@Test
+	public void testConcurrentModificationWithApplyToAllKeysAndKeyedStateLambdaFunction() throws Exception {
```

Comment:
Wasn't the goal of this test to make sure you can pass lambda? Please correct the name of the test. Maybe: `testApplyToAllKeysLambdaFunction`?